### PR TITLE
Home page search bar

### PR DESCRIPTION
**Pull request by kumuthac23**

### CODE REVIEW

File: src/pages/home/HomePageSlicker.tsx [hunk 1]
```diff
@@ -50,36 +50,52 @@ function HomePageSlicker() {
...
-      );
+  const fetchProducts = async (
+    menuId: string | undefined,
+    searchTerm: string
+  ) => {
+    try {
+      let response;
+
+      if (searchTerm.length < 3 && !menuId) {
+        response = await getProductsByMenuIdWithSearchTerm(undefined, "");
```
Comment (praveensumathi) [inline src/pages/home/HomePageSlicker.tsx:65]: why you are passing undefined, pass empty string

File: src/pages/home/HomePageSlicker.tsx [hunk 2]
```diff
@@ -50,36 +50,52 @@ function HomePageSlicker() {
...
-        }
-      );
+  const fetchProducts = async (
+    menuId: string | undefined,
+    searchTerm: string
+  ) => {
+    try {
+      let response;
+
+      if (searchTerm.length < 3 && !menuId) {
```
Comment (praveensumathi) [inline src/pages/home/HomePageSlicker.tsx:64]: it should be greater than 3 right ?

File: src/pages/home/HomePageSlicker.tsx [hunk 3]
```diff
@@ -50,36 +50,52 @@ function HomePageSlicker() {
...
+  }, [selectedMenuId, searchTerm]);
 
-            setProducts(products);
-          } else {
-            setProducts([]);
-          }
-        }
-      );
+  const fetchProducts = async (
+    menuId: string | undefined,
```
Comment (praveensumathi) [inline src/pages/home/HomePageSlicker.tsx:58]: remove undefined type from here and assign empty as default argument value

File: src/pages/home/HomePageSlicker.tsx [hunk 4]
```diff
@@ -50,36 +50,52 @@ function HomePageSlicker() {
...
-          if (response && response.data.length > 0) {
-            const products: IProductDropDownData[] = response.data.map(
-              (product) => ({
-                _id: product._id,
-                title: product.title,
-                posterURL: product.posterURL,
-              })
-            );
+    fetchProducts(selectedMenuId, searchTerm);
+    setSearchTerm("");
```
Comment (praveensumathi) [inline src/pages/home/HomePageSlicker.tsx:54]: why you use this here again?

File: src/pages/catering/SearchBar.tsx [hunk 5]
```diff
@@ -152,7 +160,25 @@ function SearchBar({ onSelectMenu, onSelectProduct }: IProps) {
...
+                {...params}
+                label="Select Food"
+                variant="outlined"
+                InputProps={{
+                  ...params.InputProps,
+                  disableUnderline: true,
+                  onChange: (event) => {
+                    const newSearchTerm = event.target.value;
+
+                    setProductValue({
```
Comment (praveensumathi) [inline src/pages/catering/SearchBar.tsx:173]: why you updating the productValue, use one state called searchterm, like in the home page, don't change the product value

File: src/services/api.ts [hunk 6]
```diff
@@ -24,18 +24,33 @@ const getAllMenus = async () => {
   }
 };
 
-const cateringfetchProductData = async (menuId = "", searchterm = "") => {
+const cateringfetchProductData = async (
+  menuId: string | undefined,
```
Comment (praveensumathi) [inline src/services/api.ts:28]: just use string type not use undefined here. assign empty as default arugument for both menuId, searchTerm

File: src/services/api.ts [hunk 7]
```diff
@@ -98,17 +113,29 @@ const getfetchProductsByMenuId = async (menuId: string) => {
 };
 
 const getProductsByMenuIdWithSearchTerm = async (
-  selectedMenuId: string,
+  selectedMenuId: string | undefined,
```
Comment (praveensumathi) [inline src/services/api.ts:116]: remove undefined type here also assign default argument value as empty for both menuId, searchTerm

File: src/pages/home/HomePageSlicker.tsx (hunk 1 above)
Comment (kumuthac23) [reply src/pages/home/HomePageSlicker.tsx:65]: changed to empty string

File: src/pages/home/HomePageSlicker.tsx (hunk 3 above)
Comment (kumuthac23) [reply src/pages/home/HomePageSlicker.tsx:58]: remove undefined

File: src/services/api.ts (hunk 7 above)
Comment (kumuthac23) [reply src/services/api.ts:116]: fixed

File: src/pages/catering/SearchBar.tsx (hunk 5 above)
Comment (kumuthac23) [reply src/pages/catering/SearchBar.tsx:173]: changed state as searchTerm

File: src/pages/home/HomePageSlicker.tsx (hunk 4 above)
Comment (kumuthac23) [reply src/pages/home/HomePageSlicker.tsx:54]: i removed in fetchProducts function.when using useEffect only working fine

**Pull request merged**